kyaml/yaml/walk/associative_sequence.go

```diff
@@ -19,21 +19,69 @@ func (l *Walker) walkAssociativeSequence() (*yaml.RNode, error) {
 		return nil, err
 	}
 
-	var key string
+	var key, strategy string
 	if l.Schema != nil {
-		_, key = l.Schema.PatchStrategyAndKey()
+		strategy, key = l.Schema.PatchStrategyAndKey()
 	}
-	if key == "" { // no key from the schema, try to infer one
+	if strategy == "" && key == "" { // neither strategy nor not present in the schema -- infer the key
 		// find the list of elements we need to recursively walk
 		key, err = l.elementKey()
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	values := l.elementValues(key)
+	// non-primitive associative list -- merge the elements
```

Comment:
can you drop a little more context here on supported values for key and strategy?